### ISSUE TITLE
Missing Spaces while using Multiple Monitors

### ISSUE THREAD

**Issue by Vision-zz** [ekalinin/SpaceName#8]
I have a dual display setup where my primary display is the Mac's display and the secondary is my External monitor. The Spaces only show the spaces from either one of the displays. If i disconnect the external monitor then this works perfectly. Otherwise it just shows the spaces from either one of the displays.

**Comment by ekalinin** [comment]
Hey,

Please, try a new version:
- https://github.com/ekalinin/SpaceName/releases/tag/0.6

**Comment by Vision-zz** [comment]
@ekalinin This works perfectly! Also a suggestion, I am not sure if this is possible since I am new to lua and hammerspoon. But showing just the desktops in that monitor would be much better. Just a suggestion that could be considered if it is programatically possible. This is great work though! Love it.

**Comment by ekalinin** [comment]
> But showing just the desktops in that monitor would be much better. Just a suggestion that could be considered if it is programatically possible. 

Good point, thanks! Will investigate whether it is possible to implement it with hammerspoon.

> This is great work though! Love it.

Thanks!

**Comment by maximneumann** [comment]
I have two monitors as well and observe this behavior: 
if "Multi Monitor Mode" is enabled, then using the SpaceName menu I can switch to any desktop, but in the menu it will show only desktop names from only one of the desktops, and I can set the name only for the desktop which is shown in the menu.

And this is determined at the start of Hammerspoon, maybe randomly. 
Basically I have to restart Hammerspoon a few times in order to show the desktop from the second monitor so that I can rename the desktops there. 
(which is a workable, but not ideal solution)

**Comment by CallaWolf** [comment]
> I have two monitors as well and observe this behavior: if "Multi Monitor Mode" is enabled, then using the SpaceName menu I can switch to any desktop, but in the menu it will show only desktop names from only one of the desktops, and I can set the name only for the desktop which is shown in the menu.
> 
> And this is determined at the start of Hammerspoon, maybe randomly. Basically I have to restart Hammerspoon a few times in order to show the desktop from the second monitor so that I can rename the desktops there. (which is a workable, but not ideal solution)

If you didn't find the solution yet, I dealt with this by reassigning the main monitor. That is, I first rename the desktops on one monitor, then set the other monitor as the primary monitor and rename the desktops there. It really works, although it is a bit tedious (it's good that it's not often necessary).
